### DISCUSSION TITLE
PSD export

### DISCUSSION THREAD

**Discussion by Yelisava** [keiwando/toonsquid#67] [Ideas]
Hello, do you have a plan to implement export as a .psd file? I can guess it might not be a trivial task, but it would broaden the field of usage of your app in bigger projects that still rely on work in photoshop. Would be increadible, really.

**Comment by keiwando** [comment]
Hi! Yes, I'm planning on adding PSD support as well. 
Are you mainly asking about PSD as a file format for the image / image sequence export or individual frames or rather the entire animation into a single PSD file using Photoshop's animation features?

**Reply by Yelisava** [reply]
Photoshop's animation features. A lot of animation/motion industry still relies on good old photoshop timeline animation for a lack of a better affordable option. Your app possesses an amazing user experience (super rare for animation apps), it's incredibly pleasant to work in, but in projects with a lot of people involved/in a certain type of pipeline/ it might be limiting not being able to have this versatility, since people have different gear. And sometimes you just want to do something in photoshop, cause you need this particular feature.
There's no need to implement those features in ToonSquid, since the fact this app is so clean is wonderful, but would be cool to have the option to delegate. I love how a similar option is made in an app (Mac/PS) called Storyboarder, where the app is quite clean and simple, having only 4-5 basic brushes and no layers, but giving you a very quirky ux the option to just send the image to photoshop or another program of your choice to refine. This was a big one for me, a very natural transition.
I don't know if the psd export as just file format would be helpful, maybe some other people could join discussion and share their views on how it might be the thing they need.

Thank you for your work!

**Reply by keiwando** [reply]
Thanks for the detailed insights!

**Reply by keiwando** [reply]
Hello again! Is it only the frame-by-frame timeline in Photoshop that is relevant to your professional use cases or also the video timeline?

**Reply by Yelisava** [reply]
Hi! It's mainly the video timeline, frame-by-frane is actually (seemingly) not so problematic, because you can just load sequence, but the struggle goes when you want to work with "instances", "drawings" that you want to last a certain amount of frames and edit them one by one. I usually end up mixing framerates during the work, especially on a rough faze. What I must also warn you as a user of this diabolical outlet (photoshop timeline) - there is a certain problem in this timeline when doing irregular frame length regarding some ehhh... empty subframes between instances. Sometimes you want to drag a frame to have a longer length and it creates a blank space between that is nor 0 or 1 frame but you see it in animation. Because of this syncing layers becomes a nightmare sometimes. I don't know who coded it but he has a special hateplace in my heart. Just sharing the experience to give a clue for you if you get stuck at some wall not understanding what goes wrong. It's not you, it's some lazy adobe monopoly.

**Reply by Yelisava** [reply]
Or do you mean by video timeline those "video layers" that have no clear instances and look like a vide brick you can draw on separate frames?

**Reply by keiwando** [reply]
I was talking about the video timeline like you described, not "video layers". Thanks for all the details and for the heads up regarding that frame length issue. If you happen to have a PSD file with this kind of an empty subframe and if you are allowed to share its contents, feel free to send me a copy of it to contact@toonsquid.com so I can investigate it in more detail and see how to best handle it during the import.
